ocaml/xapi-cli-server/cli_operations.ml

```diff
@@ -6800,6 +6800,19 @@ let pool_get_guest_secureboot_readiness printer rpc session_id params =
        (Record_util.pool_guest_secureboot_readiness_to_string result)
     )
 
+let pool_sync_bundle fd _printer rpc session_id params =
+  let filename = List.assoc "filename" params in
+  let prefix = uri_of_someone rpc session_id Master in
+  let make_command task_id =
+    let uri =
```

Comment:
This can use `Uri.make` and `Uri.to_string`.

Reply:
I replaced the uri concatenation code with below:
```
let uri =
           Uri.(
             make ~scheme:"https" ~host:master_address
               ~path:Constants.put_bundle_uri
               ~query:
                 [
                   ("session_id", [Ref.string_of session_id])
                 ; ("task_id", [Ref.string_of task_id])
                 ]
               ()
             |> to_string
           )
```
But when I upload a very tiny file, it hangs. I don't know why. So I reverted this code. I need to more investigation tomorrow for that.

Reply:
The code above just constructs the URI; hard to imagine that this causes a different behaviour.

Reply:
We need to understand what this `uri` here is supposed to be, as looking at the `uri_of_someone` function, the result of it is `""` if given `Master`. But the `Uri.make` call above always adds a `https://` prefix in addition.

Reply:
I replaced `https` with `http` then the issue was resolved. As I understand, Using `https` will send the request to stunnel and it will keep a connection so that it will hang the connection when the file is incomplete.